src/resources/resources.service.ts

```diff
@@ -2,43 +2,39 @@ import {
     BadRequestException,
     Injectable,
     NotFoundException,
-    UnauthorizedException,
 } from "@nestjs/common";
 import { CreateResourceDto } from "./dto/create-resource.dto";
 import { UpdateResourceDto } from "./dto/update-resource.dto";
 import { PrismaService } from "../prisma/prisma.service";
-import { GlobalService } from "../global/global.service";
+import { CustomRequest } from "../global/types/CustomRequest";
+import { manageOwnVoyageTeamWithIdParam } from "../ability/conditions/voyage-teams.ability";
+import { manageOwnResourceById } from "../ability/conditions/resource.ability";
 
 @Injectable()
 export class ResourcesService {
-    constructor(
-        private prisma: PrismaService,
-        private readonly globalService: GlobalService,
-    ) {}
+    constructor(private prisma: PrismaService) {}
 
     async createNewResource(
-        req,
+        req: CustomRequest,
         createResourceDto: CreateResourceDto,
         teamId: number,
     ) {
-        const { url, title } = createResourceDto;
-        const userId = req.user.userId;
-
         // make sure teamId exists
         await this.checkTeamExists(teamId);
 
-        const teamMember =
-            await this.globalService.validateLoggedInAndTeamMember(
-                teamId,
-                userId,
-            );
+        manageOwnVoyageTeamWithIdParam(req.user, teamId);
+        const { url, title } = createResourceDto;
+
+        const teamMemberId = req.user.voyageTeams.find(
```

Comment:
is there a possibility that teamMemberId is null/undefined? If yes we need to handle this, instead of passing a possible undefined below (line 50). 

Typescript says it can be `number | underfined` but `manageOwnVoyageTeamWithIdParam` should have ruled out that possiblity to be undefined? Unless the user is an admin which bypasses that check

Would it be beneficial to extract this function? Feels like it could be something useful for other services? It looks like there's another usage in features.service

Reply:
Currently , I see that there is  a `getVoyageTeamMemberId` in the `global service` . So we can use that function to find the member Id , hence it will not be undefined anymore.

Reply:
@Ajen07 good find. I don't even remember this function😂